x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts

```diff
@@ -41,6 +42,14 @@ export const postActionsConnectorExecuteRoute = (
         // get the actions plugin start contract from the request context:
         const actions = (await context.elasticAssistant).actions;
 
+        // if not langchain, call execute action directly and return the response:
+        if (!request.body.assistantLangChain) {
+          const result = await executeAction({ actions, request, connectorId });
+          return response.ok({
+            body: result,
+          });
+        }
+
```

Comment:
Did `assistantLangChain` need to be pushed server side for this fix? @andrew-goldstein intentionally kept the split in logic between calling the actions framework, and the new langchain implementation at the client API layer for separation of concerns. There's no intent to continue supporting an actions 'pass through' once this new API is validated, so unless this is somehow required for the fix I'd be hesitant to push this logic into this route.

Reply:
Looks like this is going to be necessary for the upcoming streaming work, so all good! 👍

Reply:
Yes, decided to move it new because I wanted to parse the response to continue returning a string from this endpoint, and I knew I'd be moving it for streaming anyways.